js/auth.js

```diff
@@ -0,0 +1,139 @@
+import { API_BASE_URL } from '../js/constants.js';
+/**
+ * Registers a new user.
+ * @async
+ * @function
+ * @param {string} name - The name of the user.
+ * @param {string} email - The email of the user.
+ * @param {string} password - The password of the user.
+ * @throws Will throw an error if registration fails.
+ * @returns {Promise<void>} No return value.
+ */
+export async function registerUser(name, email, password) {
+    try {
+        const response = await fetch(`${API_BASE_URL}/social/auth/register`, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({
+                name,
+                email,
+                password,
+            }),
+        });
+
+        const errorMessageDiv = document.getElementById('error-message');
+
+        if (!response.ok) {
+            const errorData = await response.json();
+            if (errorData.errors && errorData.errors[0].message === 'Profile already exists') {
+                errorMessageDiv.textContent = 'A profile with this email or username already exists. Please try logging in or use different credentials.';
+            } else {
+                errorMessageDiv.textContent = 'Registration failed. Please try again.';
+                throw new Error('Registration failed');
+            }
+        } else {
+            const data = await response.json();
+            console.log('User registered successfully:', data);
+            errorMessageDiv.textContent = 'User registered successfully!'; // Display a success message
+            errorMessageDiv.classList.remove('text-danger');
+            errorMessageDiv.classList.add('text-success');
+            window.location.href = '/profile/index.html';
+        }
+    } catch (error) {
+        console.error('Error:', error);
+    }
+}
+/**
+ * Logs in a user.
+ * @async
+ * @function
+ * @param {string} email - The email of the user.
+ * @param {string} password - The password of the user.
+ * @throws Will throw an error if login fails.
+ * @returns {Promise<void>} No return value.
+ */
+export async function loginUser(email, password) {
+    try {
+        const response = await fetch(`${API_BASE_URL}/social/auth/login`, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({
+                email,
+                password,
+            }),
+        });
+
+        if (!response.ok) {
+            throw new Error('Login failed');
```

Comment:
You can provide the user an info on what to do next when he/she fails to login successfully. Here is an example
`Login failed. Please, register as a new user`

Reply:
Yes, agreed! Done

Reply:
Thanks coach